### PR TITLE
Remove result parser and smart contract querties

### PR DESCRIPTION
**Pull request by danielailie**

### CODE REVIEW

File: src/smartContracts/smartContractController.ts [hunk 1]
```diff
@@ -16,23 +16,25 @@ import { SmartContractTransactionsFactory } from "./smartContractTransactionsFac
...
-    private queryController: SmartContractQueriesController;
     private transactionWatcher: TransactionWatcher;
     private txComputer: TransactionComputer;
+    private networkProvider: ApiNetworkProvider | ProxyNetworkProvider | INetworkProvider;
+    private abi?: AbiRegistry;
 
-    constructor(options: { chainID: string; networkProvider: INetworkProvider; abi?: AbiRegistry }) {
+    constructor(options: {
+        chainID: string;
+        networkProvider: ApiNetworkProvider | ProxyNetworkProvider | INetworkProvider;
```
Comment (andreibancioiu) [inline src/smartContracts/smartContractController.ts:26]: Cannot be only `INetworkProvider`?

File: src/smartContracts/smartContractController.ts [hunk 2]
```diff
@@ -79,17 +85,93 @@ export class SmartContractController {
...
+    parseExecute(transactionOnNetwork: TransactionOnNetwork): resources.ParsedSmartContractCallOutcome {
+        return this.parser.parseExecute({ transactionOnNetwork });
+    }
+
+    async awaitCompletedExecute(txHash: string): Promise<resources.ParsedSmartContractCallOutcome> {
+        const transaction = await this.transactionWatcher.awaitCompleted(txHash);
+        return this.parseExecute(transaction);
+    }
+
+    async query(options: SmartContractQueryInput): Promise<any[]> {
```
Comment (popenta) [inline src/smartContracts/smartContractController.ts:92]: we can use `SmartContractQuery` or simply pass the arguments.

**Review by andreibancioiu — APPROVED** [review]

File: src/abi/interaction.local.net.spec.ts [hunk 3]
```diff
@@ -172,22 +113,23 @@ describe("test smart contract interactor", function () {
...
         this.timeout(140000);
 
         let abiRegistry = await loadAbiRegistry("src/testdata/basic-features.abi.json");
         let contract = new SmartContract({ abi: abiRegistry });
-        let controller = new ContractController(provider);
+        let controller = new SmartContractController({
+            chainID: "localnet",
+            networkProvider: provider,
+            abi: abiRegistry,
+        });
```
[the remark this answers is not in the document]
Comment (danielailie) [reply src/abi/interaction.local.net.spec.ts:132]: nope, the one deleted were duplicated this is not

File: src/abi/interaction.local.net.spec.ts [hunk 4]
```diff
@@ -172,22 +113,23 @@ describe("test smart contract interactor", function () {
 
         const executeTxHash = await provider.sendTransaction(transaction);
         transactionOnNetwork = await transactionCompletionAwaiter.awaitCompleted(executeTxHash);
-        const typedBundle = resultsParser.parseOutcome(
-            transactionOnNetwork,
-            abiRegistry.getEndpoint("getUltimateAnswer"),
-        );
+        const responseD = queryController.parseExecute(transactionOnNetwork);
```
[the remark this answers is not in the document]
Comment (danielailie) [reply src/abi/interaction.local.net.spec.ts]: done

File: src/abi/interaction.local.net.spec.ts [hunk 5]
```diff
@@ -172,22 +113,23 @@ describe("test smart contract interactor", function () {
...
 
-        assert.lengthOf(typedBundle.values, 1);
-        assert.deepEqual(typedBundle.firstValue!.valueOf(), new BigNumber(42));
-        assert.isTrue(typedBundle.returnCode.equals(ReturnCode.Ok));
+        assert.isTrue(response.contracts.length == 1);
+        assert.deepEqual(responseD.values[0], new BigNumber(42));
+        assert.isTrue(response.returnCode == "ok");
```
[the remark this answers is not in the document]
Comment (danielailie) [reply src/abi/interaction.local.net.spec.ts]: done

**Review by popenta — APPROVED** [review]

**Pull request merged**